### PR TITLE
Use conditional conversation nodes to simplify some conversation

### PR DESCRIPTION
**Pull request by warp-core**
## Summary
Allowed talking to Foster and all the Hai researchers when examining the Emerald Sword.
Simplified a Skadenga conversation.
Simplified a Patir conversation.
Hide already selected options in the culture conversation with Taely at the end of the visit to a Paradise world mission.
Simplify the Remnant Celebration conversation. Text is no longer duplicated between the annotated and unannotated paths.
Also, one version of a line in the Celebration conversation said `toward` while the other said `towards`. When with `towards` for the unified version but that might not be correct.

### CODE REVIEW

File: data/human/deep missions.txt [hunk 1]
```diff
@@ -2825,21 +2825,33 @@ mission "Home for Skadenga 1"
...
+					to display
+						not "skadenga: asked about researchers"
 				`	"Is it safe to drive over the ice?"`
+					to display
+						not "skadenga: asked about ice"
 					goto ice
 				`	"Are there sea-monsters beneath the ice?"`
+					to display
+						not "skadenga: asked about monsters"
 					goto monster
```
Comment (UnorderedSigh) [inline data/human/deep missions.txt:2837]: Putting the `goto` before `to display` is more readable since all the text related to one option is lined up. If the `goto` is after `to display` then it looks like it is associated with the next choice.

```suggestion
					to display
						not "skadenga: asked about researchers"
				`	"Is it safe to drive over the ice?"`
					goto ice
					to display
						not "skadenga: asked about ice"
				`	"Are there sea-monsters beneath the ice?"`
					goto monster
					to display
						not "skadenga: asked about monsters"
```

I know from a programming perspective, you check if you display it before goto-ing to it, but from a readability prespective, the opposite is easier to follow.

File: data/sheragi/archaeology missions.txt [hunk 2]
```diff
@@ -1363,14 +1363,23 @@ mission "Sheragi Archaeology: The Emerald Sword 3"
 			`	After patching as many micrometeorite holes as you can find, you use your ship's own life support to restore the Emerald Sword's atmosphere, and connect your ship's power to it to get the ship's systems minimally functional. Once it's in breathable condition, the crew boards the ship and begins getting to work. With the freshly pressurized atmosphere, you can't help but think that it smells like a tomb.`
 			`	Foster divides the crew up into various groups to explore the massive vessel and report back with their findings. "We may be here for a while. There's no telling how long it will take us to get this thing operational again. It has been many thousands of years after all," he says. Foster immediately begins inspecting one of the many Sheragi remains on board, while Eyama, Rakei, and Desyo each explore the ship with their own groups.`
 			`	Within half an hour, the entire ship is swarming with crew members hard at work studying the ship (and the remains of its prior inhabitants) and attempting to get it operational.`
+			label exploration
 			choice
 				`	(Look for Foster to see what he's doing.)`
+					to display
+						not "spoke with foster"
```
Comment (UnorderedSigh) [inline data/sheragi/archaeology missions.txt:1370]: Endless Sky has only a global namespace, so you should consider having more text before this (like "Sheragi Archaeology:") to prevent future name clashes with other stories.

File: data/human/deep missions.txt (hunk 1 above)
Comment (UnorderedSigh) [reply data/human/deep missions.txt:2837]: Look at the massive conversation in "Hai Reveal [A09] History of Appeasement" in this PR to see a larger version. It may be more apparent there. It's around line 1040 of this:

https://github.com/endless-sky/endless-sky/pull/7658/files#diff-73fb5ebd354e9467bdc3d6b9d357960d835f9e9740aabdab0d3197a389076d7e

Comment (warp-core) [reply data/human/deep missions.txt:2837]: @Amazinite has suggested empty lines between the choices in complex blocks to help separate them.

We both prefer to list the conditions first because, when reading the data, whether or not an option should be displayed is probably going to be relevant before the `goto`, since the `goto` doesn't matter if it's not being displayed.

Comment (UnorderedSigh) [reply data/human/deep missions.txt:2837]: That is not what is in the wiki. It has the "goto" before the "to display"

```
    choice
        <text>
            [<endpoint> | goto <label>]
            ["to display"]
                <condition>
```

Comment (UnorderedSigh) [reply data/human/deep missions.txt:2837]: If it is written that way, then your eyes will quickly pick out both the goto and "to display," so there's really no issue of which one is first.

**Comment by Zitchas** [comment]
To be honest, I'm not really a fan of the whittling down of conversational options in situations where the player/protagonist could realistically repeat themselves. In other RPGs with conversations (the Mass Effect and Baldur's Gate serieses, for instance) I often revisited questions, especially later on in big conversations.

Tldr: If the protagonist could repeat the question, let them.

**Comment by UnorderedSigh** [comment]
> If the protagonist could repeat the question, let them.

While reading through Hai Reveal, I think there's a mix of situations, where it's just casual conversation, and when it is critical information. For casual conversation, the question coming back just looks sloppy, and serves no real purpose. For critical information, I think it would look better if the question varied slightly.

First time around: "How do I capture Widgets from the Enemy?"
Second time: "Can you tell me again how I can capture Widgets from the Enemy?"

This way, the game doesn't feel like it's mindlessly repeating the same dialog, but rather, responding to the player.

**Comment by UnorderedSigh** [comment]
Here's an example of a conversation with looping dialog, where the question changes if you're requesting the information a second time:

https://github.com/endless-sky/endless-sky/pull/7594

In one case, the response changes slightly (blocking vs. blocking2).

**Comment by Amazinite** [comment]
> In other RPGs with conversations (the Mass Effect and Baldur's Gate serieses, for instance) ...

Being able to ask the same question multiple times in a game like Mass Effect makes some sense because you don't have some audio transcript (as far as I'm aware) that you can pull up to see what was said, so the only way to double check what was said is to ask again. In Endless Sky we can just scroll up to see what was previously said in a conversation.

That is to say, repeating a question in a game like Mass Effect is a thing because that game's medium where dialog is spoken lacks the ability to review what was said without repeating the dialog. But in Endless Sky where dialog is written, we don't have that issue. We shouldn't implement a solution to a problem we don't have.

**Comment by Zitchas** [comment]
Perhaps, but it also doesn't change the fact that fit many of these conversations, repeating oneself is perfectly natural. Much more natural and immersive than rewinding time.

What would be much more valuable would be to allow options to be shared or as they are picked so it is clear that they are repeats while still allowing the character to pick them.

**Comment by warp-core** [comment]
> Perhaps, but it also doesn't change the fact that fit many of these conversations, repeating oneself is perfectly natural. Much more natural and immersive than rewinding time.

Natural how? If I were having a conversation like these with someone irl and I repeated my question after they asked it, I would be somewhat concerned if their response was identical, there would probably be some reaction to the reputation of the question, especially if I'm doing it many times. If I just wanted them to repeat the answer, I wouldn't repeat the whole question anyway, I'd ask them to repeat their answer to the last question, or the one before that, or "to the question about <X>", or just "Could you repeat what you said about <X>?"

Nothing about repeating your question exactly and getting the exact same response in a conversation with a real person would be natural.

**Comment by Zitchas** [comment]
I agree that, ideally, every instance would be different. I don't think it is worth that much work, though. That being said, being able to repeat a question allows more freedom and is more natural than blocking out off. Being able to ask something a second time and get the same response is more natural than having that question permanently remove from possibility.

Also, while Amazinite's comment about Mass Effect being an audio game is true, Baldur's Gate series was not. I've could easily rewind the conversation there, but being able to simple trust conversational choices, where appropriate, was much more useful and had better conversational flow.

**Review by Zitchas — DISMISSED** [review]
In terms of naturalness, I'll point to my own experience, in rough categories. Please note these are not intended to be generalizations or stereotypes, just statements of my experience:
1. I know quite a few older people, most of whom are still quite intellectually fit, who nonetheless will repeat themselves, both in retreading statements they made earlier in the conversation, and asking questions that have already been answered earlier. The longer the conversation, the more likely they are to repeat something.
2. I also know a number of younger people who suffer from various forms and intensities of attention, hyperactivity, and related disorders. Some diagnosed, some not. Many of these will repeat statements and questions multiple times in a conversation, sometimes before even giving the person they are speaking with the chance to finish their response.
3. I know a few people who suffer from traumas of various kinds, typically things related to head injuries, vehicle accidents, or serious illness. In many of these cases it has been medically documented that, despite their brain otherwise being relatively healthy and functional, their memory suffers enough that they tend to repeat statements and questions on a not-infrequent basis.
4. I also know a number of people who do not fall into either of those two groups who are are just frequently tired, sick, or otherwise not functioning at their best. These people also often "zone out" during conversations, and have to repeat questions to get the full picture.

In all these cases responses such as "What, weren't you paying attention, I said X" or "I already answered that" are just going to generate hurt feelings and frustration; so they virtually always get a functionally identical response with little more than minor variations, depending on the circumstances. Likewise, ignoring their question or just telling them to "remember better"(functionally what is being attempted here by removing the opportunity to re-ask a question and/or telling them to go reread the past conversation) is both insulting and un-natural to everyone in these 4 groups of people. Right up there with telling someone with broken arms to "just lift harder" to carry a heavy and awkward piece of furniture.

As such, while simplification is good, I do not agree with removing people's opportunity to repeat questions, as I equate that with discrimination against people from these situations.

Thus my request for change here would be that this PR focus purely on simplifying conversations, as the the title says, and remove the removal of people's opportunity to repeat themselves. If you wish to have a larger debate on that, I'd suggest moving the removal of people's ability to repeat themselves to a stand-alone PR that does that and only that, and continue the discussion there.

**Comment by warp-core** [comment]
Draft because I'm not actually sure these things work properly.

**Comment by warp-core** [comment]
https://github.com/endless-sky/endless-sky/commit/74e665f59af9e2cf05cf0fccd15b42e1c8517035 should fix the weirdness I encountered.

**Comment by warp-core** [comment]
@Zitchas this no longer changes the behaviour of the conversation in the Celebration mission, only removes the need for many almost identical choice blocks, and the repetition of the entire event conversation.

File: data/human/deep missions.txt (hunk 1 above)
Comment (Terin) [reply data/human/deep missions.txt:2839]: I definitely had a hard time picking out which lines the `to display` were associated with at first. For me personally, I think I'd like that _first_, followed by the dialog line, then the goto. But that's probably not doable with the code as-is. But either way, when it's getting to 3+ lines of text, breaking it up with newlines does feel like the way to go, yeah.

**Review by Terin — APPROVED** [review]
Outside of potentially breaking up the conditional dialog lines with newlines to make them more parseable, looks great to me. Very glad to have this functionality in use now.

**Comment by Amazinite** [comment]
The ability to remove choice loops using `to display` nodes is something that was talked about frequently prior to them being merged. It was one of the big reasons people wanted the feature. Even you, Zitchas, gave a +1 to one of my comments referring to removing choice loops from one of your missions after #5440 was merged (https://github.com/endless-sky/endless-sky/pull/6655#issuecomment-1089898714). Now if for your own missions you want to be able to have choice loops then that's fine, but I don't think that's something we should enforce on the entire game, and I will maintain that we should avoid choice loops wherever possible.
